### DISCUSSION TITLE
Positional arguments with BaseModel

### DISCUSSION THREAD

**Discussion by IansGithubAcc** [pydantic/pydantic#6510] [Question]
I'm trying to make positional arguments work with the BaseModel.
Whether or not using positional arguments is good practice is debatable, but regardless this is what I am trying to accomplish.

One option is to write a decorator which fill in the positional arguments as kwargs, however this does not work well with static type checking.

The metaclass used by BaseModel `ModelMetaclass`, uses the `@dataclass_transform` decorator from `typing_extensions` to forbid the use of positional arguments. Simply changing the `kw_only_default` arguement here seems to do the job without any issues, but I do not want to edit the Pylance code for obvious reasons.

Lastly, I thought to change the `__dataclass_transform__` attribute of `BaseModel.__class__` after importing it. Sadly, this does not work since the static intepreter has already read the boolean as `True` during the import.

Any ideas or insight would be greatly appreciated.

**Comment by GugelRobin** [comment]
Hello @IansGithubAcc
I've had a similar issue and what I did was setting the `kw_only` arg to false in the `Field` function (aka `foo: int = Field(kw_only=False)` - this works for very well for pylance. 

As far as I know there's sadly no way to change `kw_only_default` to false - even when you'd apply `__dataclass_transform__(kw_only_default=False)` to your decorator it seems like pylance rather uses the metaclasses dataclass-transform definition than the one added by the decorator.

I've submitted a feature request to enable positional arguments using `= Field(kw_only=False)`.
#6792

**Comment by samuelcolvin** [comment]
I would argue you should use a dataclass, or `validate_call`.

Otherwise I don't think it's a good idea and I doubt you can get it to play nicely with static typing.

**Reply by IansGithubAcc** [reply]
Thanks for the reply! How would I use `validate_call` to achieve this? I'm unfamiliar with the decorator, but it looks like a convenient way of applying configs to methods/functions. However, `kw_only_default` is not present in the standard config.
